pandas/core/generic.py

```diff
@@ -10658,6 +10676,12 @@ def shift(
         if periods == 0:
             return self.copy(deep=None)
 
+        if is_list_like(periods) and len(self.shape) == 1:
+            return self.to_frame().shift(
+                periods=periods, freq=freq, axis=axis, fill_value=fill_value
+            )
```

Comment:
If I'm following this right, the call stack will be:

 - NDFrame.shift
 - DataFrame.shift
 - NDFrame.shift
 - Series.shift

I think it would be better to handle this directly in Series.shift - is that possible?

Reply:
No, because there is no `Series.shift` (unless I'm missing something). E.g. nothing in `pandas/core/series.py`.

Reply:
https://pandas.pydata.org/docs/reference/api/pandas.Series.shift.html

Reply:
I see - Series doesn't override; that makes this more palatable. Thanks.